### ISSUE TITLE
Regarding Fine Tuning SAM Decoder

### ISSUE THREAD

**Issue by qevr** [facebookresearch/segment-anything#505]
Hi,
I have a list of images and multiple bounding boxes and masks per image . For this I want to fine tune segment anything model mask decoder, but the confusion I have is I want to calculate image embedding for one image only once but I want to calculate loss for predicted_masks and gt_masks for every bounding box . If , I follow this approach then how to back propagate using pytorch trainer ? Any code link some one can share?

I don't want to calculate image embedding for same image very time I generate mask for a bounding box and backpropagate for every loss generated from bounding boxes instead of accumulating loss from bounding box for an image and then go backwards.

**Comment by Geoyi** [comment]
Following the threads. 
Mine are points not bounding boxes and I have the similar question if I understand yours correctly.
